### PR TITLE
Json homework person&weather

### PR DESCRIPTION
**Pull request by Evgenygroup**
#1

### CODE REVIEW

File: JsonRead/src/main/java/weather/Demo.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.net.URL;
+
+public class Demo {
+    public static void main(String[] args) throws Exception {
+
+        String city = "Berlin,de";
+        String uRLString = "https://api.openweathermap.org/data/2.5/" +
```
Comment (sergeylukichev) [inline JsonRead/src/main/java/weather/Demo.java]: Think of how to make the application reusable and testable. Hints: define classes, which serve different purposes: one uses object mapper to read from the internet, another one deals with outputting wether in different formats, etc.

Comment (Evgenygroup) [reply JsonRead/src/main/java/weather/Demo.java]: more or less so?

File: JsonRead/src/main/java/weather/Main.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public class Main {
+
+    public static void main(String[] args) throws IOException {
+
+        String city = "Berlin,de";
+        String uRLString = "https://api.openweathermap.org/data/2.5/" +
+                "weather?q=" + city + "&units=metric&appid=335838461bf85dd808aef371792bb429";
+        
+        ReadFrom read = new ReadFromInternet();
+        ForecasterService forecasterService = new ForecasterService();
```
Comment (sergeylukichev) [inline JsonRead/src/main/java/weather/Main.java]: If ReadFrom needs the ForecastService, the pass ForecastService to the ReadFromInternet as parameter to constructor.

File: JsonRead/src/main/java/weather/ReadFromInternet.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.function.Function;
+
+public class ReadFromInternet implements ReadFrom {
+
+    public WeatherForecast readFrom(String str) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (sergeylukichev) [inline JsonRead/src/main/java/weather/ReadFromInternet.java]: You do not need to create ObjectMapper every time you want to readFrom(). It should be a dependency (field) of ReadFromInternet class, created either in the constructor or passed as a parameter to constructor (Inversion of Control).

File: JsonRead/src/main/java/weather/Demo.java (hunk 1 above)
Comment (sergeylukichev) [reply JsonRead/src/main/java/weather/Demo.java]: URLString is a property of a class, which is responsible for getting data from the Internet. city on the other hand, is a parameter, which denotes the city I need to know the weather in. As user, I do not need (and should not) know which server/URL is used to get the data from. As well as your api id should not be visible to clients.

**Comment by sergeylukichev** [comment]
better now

**Comment by Evgenygroup** [comment]
Thanks

**Pull request merged**